### PR TITLE
Vyu16 nag compilation

### PR DESCRIPTION
**Pull request by william-dawson**
This pull request includes @vyu16 's modifications for NAG compilation. He also mentioned that `sizeof` cannot be used with NAG, so I have replaced those calls with MPI calls.

(It looks like I messed up the procedure for updating a pull request, so here's a new one...)

### CODE REVIEW

**Comment by william-dawson** [comment]
@vyu16 can you check if these modifications are sufficient to allow compilation with NAG?

**Comment by vyu16** [comment]
@william-dawson  It worked with nagfor 6.2, which is the only version I could test.

File: Source/Fortran/MatrixReduceModule.F90 [hunk 1]
```diff
@@ -1,8 +1,7 @@
 !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 !> Module for reducing matrices across processes.
 MODULE MatrixReduceModule
-  USE DataTypesModule, ONLY : NTREAL, MPINTREAL, NTCOMPLEX, MPINTCOMPLEX, &
-       & MPINTINTEGER
+  USE DataTypesModule, ONLY : NTREAL, MPINTREAL, MPINTCOMPLEX, MPINTINTEGER
   USE SMatrixAlgebraModule, ONLY : IncrementMatrix
```
[the remark this answers is not in the document]
Comment (vyu16) [reply Source/Fortran/MatrixReduceModule.F90]: Actually, never mind... I was reading an old file -_-!

**Comment by william-dawson** [comment]
Thanks a bunch @vyu16 . I particularly appreciate you finding the unused "use" statements. If you find any redundant or useless code, feel free to remove it, the less code the better!

**Pull request merged**